### ISSUE TITLE
How to set adequate threshold for different networks?

### ISSUE THREAD

**Issue by zyxuaan** [Jackn0/snn_optimal_conversion_pipeline#2]
Hello, I really ​appreciate your work, but there's a little confusion for me about how to set the threshold to reproduce the test accuracy of the experimental results in your paper.

Configs for training and simulation are in the [settings.py](https://github.com/Jackn0/snn_optimal_conversion_pipeline/blob/main/models/settings.py), including the setting of thresh.

`parser.add_argument('--thresh', default=0, type=int, help='relu threshold, 0 means use common relu')`

I notice that the default thresh is set to be 0, but how can I set an adequate value(by using the technique of  "ReLU +threshold + shift") to get a better result?

**Comment by Jackn0** [comment]
Hello, hoping this helps. You can download the pre-trained model and load it by changing its model_save_name in main_simulation.py. For example, if you want to test the model of VGG16 on CIFAR10, after downloading the model "vgg16-cifar10-relu1.pth: and changing the model_save_name, you may run the main_simulation.py by command:
`python main_simulation.py --thresh 1 --T 64 --shift_snn 64`

BTW, if you want to train an ANN model with threshold ReLU, you may first initialize the weight and bias of the convolution layers as the paper have mentioned, such as:
`for m in self.modules():
    if isinstance(m, nn.Conv2d):
        nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity='relu')`
The running command is:
`python main_train.py --dataset CIFAR10 --arch VGG16 --batch_size 512 --learning_rate 0.1 --epochs 300 --thresh 1 --init_epoch 140`
I have tested the command with random seed 1005 and saved the training log and pre-trained model to the corresponding google driver (cifar10/vgg16).

**Comment by zyxuaan** [comment]
Thx a lot for your advice in detail！
